### ISSUE TITLE
Disable account creation if user exists in single-user mode

### ISSUE THREAD

**Issue by csarven** [solid/node-solid-server#1369]
In single-user mode, if config has multiuser.false and there is already a user, disable (4xx) /api/accounts/new , /register etc.

**Comment by michielbdejong** [comment]
Is this a duplicate of #1368?

**Comment by csarven** [comment]
Perhaps. The idea here is that once the first (and only) user is created in single-user mode, anything pertaining to account creation should be prohibited. If user account is tracked through profile/card or whatever is a separate issue I think.

**Comment by michielbdejong** [comment]
> there is already a user

how would you track that? if you don't change how it's tracked (i.e. keep using presence of profile/card as the check), then it doesn't matter much if you only disable the actual api call (as is currently the case) or also disable the /register route

**Comment by csarven** [comment]
I don't know about the internals of NSS. If for instance once the first and only user is created, all potentially sensitive or useless paths need not be exposed eg. wikis, cms.. clean up once the setup is complete. It just closes/reduces attack surface.

If checking for the existence of `pathCard` value (on disk?) is the way to determine if a user was created and still exists, then I suppose the rest follows.

**Comment by michielbdejong** [comment]
Ah so you mean instead of exposing the /api/accounts/new endpoint and then doing the check [here](https://github.com/solid/node-solid-server/blob/master/lib/requests/create-account-request.js#L149), you could edit the actual router configuration so that that route is not even configured (i.e. would basically be a 404 instead of the 400 that it currently returns).

I can see how that would make sense in for instance a php admin tool, where each route corresponds to a `.php` file. But in node, it feels cleaner to keep the route, and then do that check in the code. Note that it has a [list of checks](https://github.com/solid/node-solid-server/blob/master/lib/requests/create-account-request.js#L147-L149) it does before creating an account, and account existence is just one of them.

**Comment by csarven** [comment]
Sure. Whether the resource accepting new account creation requests (eg. api/accounts/new ) is accessible is arguably an implementation detail. (Personally I'd hide/forbid that resource from the Web.) I would just add that the status code for the response should probably be the same for all implementations. Do you know if this is document/spec'd somewhere?

**Comment by michielbdejong** [comment]
OK, so let's leave it as it is, then?

> Do you know if this is document/spec'd somewhere?

This is interaction between user and pod provider, not between app and pod, so none of that is (currently) standardized. Each pod server implementation can choose its own routes and semantics.
